src/App.tsx

```diff
@@ -1,14 +1,67 @@
-/* eslint-disable max-len */
-import React from 'react';
+import React, { useCallback, useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
-import '@fortawesome/fontawesome-free/css/all.css';
-
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+import { FilterTypes } from './components/enums/FilterTypes';
+
+const filterAndSearchTodos = (
+  todos: Todo[],
+  filterQuery: FilterTypes,
+  searchQuery: string,
+): Todo[] => {
+  const filteredTodos = todos.filter(todo => {
+    if (filterQuery === FilterTypes.All) {
+      return true;
+    }
+
+    return filterQuery === FilterTypes.Completed
+      ? todo.completed
+      : !todo.completed;
+  });
+
+  if (searchQuery) {
+    return filteredTodos.filter(todo =>
+      todo.title.toLowerCase().includes(searchQuery.toLowerCase()),
+    );
+  }
+
+  return filteredTodos;
+};
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [filterQuery, setFilterQuery] = useState(FilterTypes.All);
+  const [searchQuery, setSearchQuery] = useState('');
+
+  useEffect(() => {
+    getTodos()
+      .then(setTodos)
+      .catch(error => {
+        // eslint-disable-next-line no-console
+        console.error('Failed to fetch todos:', error);
+      });
+  }, []);
+
+  const onCloseHandler = useCallback(() => {
+    setSelectedTodo(null);
+    setSearchQuery('');
+  }, []);
+
+  const handleFilterChange = useCallback((filterOption: FilterTypes) => {
+    setFilterQuery(filterOption);
+  }, []);
+
+  const handleInputChange = useCallback((input: string) => {
+    setSearchQuery(input);
+  }, []);
```

Comment:
These functions don't make much sense as they just pass arguments to the next function